src/main/scala/chiseltest/package.scala

```diff
@@ -52,7 +53,53 @@ package object chiseltest {
       }.foreach { case (k, v) =>
         v match {
           case record: Record => record.expectPartial(value.elements(k).asInstanceOf[Record])
-          case data:   Data   => data.expect(value.elements(k))
+          case data: Data => data.expect(value.elements(k))
+        }
+      }
+    }
+  }
+
+  implicit class testableVec[T <: Vec[_]](x: T) {
+
+    /** Poke the given signal with a [[Vec.litValue()]]
+      * Literals of this Record can be instantiated with
+      * {{{
+      *   someVec.Lit(_.elements("foo") -> 4.U)
+      * }}}
+      * `pokePartial` will only poke [[Input]] signals of `x`,
+      * and elements of `x` which contain no literal will be ignored.
+      */
+    def pokePartial(value: T): Unit = {
+      require(DataMirror.checkTypeEquivalence(x, value), s"Vec type mismatch")
+      x.getElements.zipWithIndex.filter { case (v, index) =>
+        DataMirror.directionOf(v) != ActualDirection.Output && {
+          value.getElements(index) match {
+            case _: T => true
+            case data: Data => data.isLit()
+          }
+        }
+      }.foreach { case (v, index) =>
+        v match {
+          case vec: T => vec.pokePartial(value.getElements(index).asInstanceOf[T])
```

Comment:
Does it make sense to match on T, as opposed to Vec?
Also, would this support Vecs inside Records and vice versa?

Reply:
Because of the parameterization of `Vec`, this seems to be the easiest correct syntactic construction
It does support rec in vec and vice versa

Reply:
But will this recursively pokePartial with an inner Bundle? It looks like it will do a normal strict poke.
Which is fine for now, to allow a recursive pokePartial later would be strictly an API expansion I think,